### PR TITLE
Fixed problem with regex in es6 import challenge

### PR DESCRIPTION
**Pull request by thecodingaviator**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

Fixed issue #17816

### CODE REVIEW

**Comment by thecodingaviator** [comment]
@Asjas

**Comment by alanpaulprice** [comment]
@thecodingaviator I suggest updating the regex to this:

`import\s+subtract\s+from\s+("|')\.\/math_functions\s*\1`

As the file being targeted is in the same directory, it is necessary to have the file name preceded by `./`.

I've also removed the need for a semi colon and allowed for spaces after the file name (with the quotes).

**Comment by thecodingaviator** [comment]
@alanpaulprice Hmm that looks better, but the semi-colon is necessary, else the next lines of code won't work when there are more.

**Comment by alanpaulprice** [comment]
@thecodingaviator I tested it without any semi colons in either file and it worked fine:

image

**Comment by thecodingaviator** [comment]
Okay, now I’m not sure if I’m wrong or it’s compiling and then running as you’re using babel. I’ll have a loot again at this tomorrow

**Comment by thecodingaviator** [comment]
I'm not sure yet so, @moT01 can you have a look?

**Comment by lynxlynxlynx** [comment]
Why wouldn't a newline be enough?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/import-a-default-export.english.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ In the following code, please import the default export, <code>subtract</code>,
 ```yml
 tests:
   - text: Properly imports <code>export default</code> method.
-    testString: getUserInput => assert(getUserInput('index').match(/import\s+subtract\s+from\s+"math_functions"/g), 'Properly imports <code>export default</code> method.');
+    testString: 'getUserInput => assert(getUserInput("index").match(/import\s+subtract\s+from\s+("\.\/+math_functions"|'\.\/+math_functions')+\s{0,}?;/g), "Ensure that the format is correct and you are import the <code>subtract</code> export from <code>"math_functions"</code>");'
```
Comment (lynxlynxlynx) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/import-a-default-export.english.md]: The explanation needs to be fixed, as now it doesn't make much sense due to the errors.

**Review by lynxlynxlynx — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/import-a-default-export.english.md (hunk 1 above)
Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/import-a-default-export.english.md]: I've made a change, check it

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/import-a-default-export.english.md]: Yeah, but I've seen a mixture of those here of FCC. Don't know what I should use. Ready to change if someone tells me so, but this should be fine too

**Comment by lynxlynxlynx** [comment]
Rereading this, why not just check for ["'] instead of duplicating the expression (like alanpaulprice suggested)? Also, this needs to be tested manually just to make sure the single quotes don't need to be escaped.

**Comment by thecodingaviator** [comment]
Nope, I'm saying it should not work if the user does that. What you and @alanpaulprice are suggesting is to check `'|"` which would not help us in checking if the closing quote is ' or " . The user might mix it up and if we implement your solution to use `'|"` we cannot check mixes

**Comment by thecodingaviator** [comment]
My very bad. You're right. I will change it to that. Sorry for the problems

**Review by lynxlynxlynx — APPROVED** [review]

**Comment by lynxlynxlynx** [comment]
Good, thanks! I know you just copied it, but is the \s* really appropriate? I don't think many people would add a space or three, but then it would be looking for the wrong filename and fail the import.

**Pull request merged**